android/src/main/java/com/robinpowered/RNMDMManager/RNMobileDeviceManagerModule.java

```diff
@@ -80,7 +125,51 @@ public void getConfiguration(final Promise promise) {
             }
             promise.resolve(data);
         } else {
-          promise.reject(new Error("Managed App Config is not supported"));
+            promise.reject(new Error("Managed App Config is not supported"));
+        }
+    }
+
+    @ReactMethod
+    public void isAutonomousSingleAppModeSupported(final Promise promise) {
+        promise.resolve(isLockStatePermitted());
+    }
+
+    @ReactMethod
+    public void isSingleAppModeEnabled(final Promise promise) {
+        promise.resolve(isLockState());
+    }
+
+    @ReactMethod
+    public void isAutonomousSingleAppModeEnabled(final Promise promise) {
+        boolean lockedStatus = false;
+        ActivityManager am = (ActivityManager) getReactApplicationContext().getSystemService(Context.ACTIVITY_SERVICE);
+        try {
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
```

Comment:
Wouldn't be against moving the _logic_ into a private method that the ReactMethod calls -- keeps a separation between implementation details from the React bindings and resolvers.

Just like `isLockState`